client/network/src/lib.rs

```diff
@@ -454,3 +454,72 @@ async fn wait_to_announce<Block: BlockT>(
 		);
 	}
 }
+
+#[derive(Clone)]
+pub struct WaitForParachainTargetBlock<Block> {
+	phantom: PhantomData<Block>,
+}
+
+impl<Block: BlockT> WaitForParachainTargetBlock<Block> {
+	/// Get warp sync target block
+	pub async fn warp_sync_get(
+		para_id: ParaId,
+		relay_chain_interface: Arc<dyn RelayChainInterface>,
+	) -> Result<oneshot::Receiver<Block::Header>, BoxedError>
+	where
+		Block: BlockT + 'static,
+	{
+		let (sender, receiver) = oneshot::channel::<Block::Header>();
+		Self::wait_for_target_block(sender, para_id, relay_chain_interface).await;
+		return Ok(receiver)
+	}
+
+	async fn wait_for_target_block(
+		sender: oneshot::Sender<Block::Header>,
+		para_id: ParaId,
+		relay_chain_interface: Arc<dyn RelayChainInterface>,
+	) {
+		let is_syncing = relay_chain_interface
+			.is_major_syncing()
+			.await
+			.map_err(|e| {
+				tracing::error!(target: LOG_TARGET, "Unable to determine sync status. {}", e)
+			})
+			.unwrap_or(false);
+
+		loop {
+			if !is_syncing {
```

Comment:
This is an endless loop when `is_syncing` is true.

Reply:
We also need to ensure that this isn't a hot loop. So either add some kind of async sleep or we come up with some different kind of solution. First we could for example wait for a block import notification (block import notifications should not arrive when we are doing a major sync). After the first block import notification
we then need to check of `is_major_sync` and if that still returns true, we should fall back to the "sleep solution" I described above.

Reply:
The loop now only starts once we receive a block import notification. 

Added a sleep described above. Will resolve if acceptable 👍